spec/models/document_spec.rb

```diff
@@ -132,4 +149,21 @@
       expect(subject.original_document).to eq(grandparent)
     end
   end
+
+  describe "#destroy" do
+    context "when document has children" do
+      let(:grandparent) { create(:document) }
+      let(:parent) { create(:document, parent: grandparent) }
+
+      before do
+        grandparent.save
+        parent.save
```

Comment:
Don't need these two saves.

It's quite rare to see `.save` in a before block in a spec. Usually you just create the object in the way you want:

- `create` if you want it saved (it's analogous to ActiveRecord's `create`)
- `build` or `described_class.new` if you don't

In the case of this spec, `subject` was created with `described_class.new` so a `subject.save` in only some `describe` blocks make sense. You shouldn't need to do it for anything else, though.